compiler/semantic/op.go

```diff
@@ -1021,25 +1028,29 @@ func (a *analyzer) semOpExpr(e ast.Expr, seq dag.Seq) (dag.Seq, error) {
 	}), nil
 }
 
-func (a *analyzer) isBool(e dag.Expr) bool {
+func (a *analyzer) isBool(e dag.Expr, strict bool) bool {
 	switch e := e.(type) {
 	case *dag.Literal:
 		return e.Value == "true" || e.Value == "false"
 	case *dag.UnaryExpr:
-		return a.isBool(e.Operand)
+		return a.isBool(e.Operand, strict)
 	case *dag.BinaryExpr:
 		switch e.Op {
 		case "and", "or", "in", "==", "!=", "<", "<=", ">", ">=":
 			return true
+		case "[":
+			return !strict
 		default:
 			return false
 		}
+	case *dag.This:
+		return !strict
```

Comment:
@nwt what about changing `isBool` to do something like this? Seems to work but I'm not sure if this covers all cases.

Reply:
I think this is the right direction. I too am not sure if this covers all the cases (but we need to be before we merge something like this).